osrf_pycommon/process_utils/get_loop_impl.py

```diff
@@ -23,25 +23,34 @@ def get_loop_impl(asyncio):
     if getattr(_thread_local, 'loop_has_been_setup', False):
         return asyncio.get_event_loop()
     # Setup this thread's loop and return it
-    if os.name == 'nt':
-        try:
-            loop = asyncio.get_event_loop()
+    try:
+        loop = asyncio.get_running_loop()
+        have_running_loop = True
+    except RuntimeError:
+        # Thrown when there is no running loop
+        have_running_loop = False
+    except AttributeError:
+        # Thrown when on Python < 3.7
+        loop = asyncio.get_event_loop()
+        have_running_loop = True
```

Comment:
@Voldivh On the other PR, you said:

```
I believe this section of the code should also look like this. However, I noticed that at this point of the code the method returns a RuntimeError which shouldn't happen according to my understanding. It would seem there is a bug at setting a new_event_loop to asyncio. Let me know your thoughts @clalancette
```

I'm not sure I understand where the problem is, and what we would need to do to fix it.  Can you explain more about where you are seeing a `RuntimeError`?